arch/arm/core/fault.c

```diff
@@ -640,11 +662,24 @@ static u32_t _FaultDump(const NANO_ESF *esf, int fault)
  * Note: exc_return argument shall only be used by the Fault handler if we are
  * building Secure Firmware.
  */
-void _Fault(const NANO_ESF *esf, u32_t exc_return)
+void _Fault(NANO_ESF *esf, u32_t exc_return)
 {
 	u32_t reason = _NANO_ERR_HW_EXCEPTION;
 	int fault = SCB->ICSR & SCB_ICSR_VECTACTIVE_Msk;
 
+#ifdef CONFIG_USERSPACE
```

Comment:
As far as I understand, only memory protection faults could fall into this category of errors. So, MemMg for ARM and BusFaults for NXP, so I wonder if we could move the code inside the respective handlers.

This comment is related to my other comment below.

Reply:
I think this would require significantly restructuring the code in fault.c.

We don't really have handlers in here, just a bunch of stuff which prints out information to the console. There's no leeway for policy decisions to be made in the way it is structured now. All the "handlers" like _MpuFault are called by FAULT_DUMP, which only calls _FaultDump() if CONFIG_FAULT_DUMP > 0 and all the return values are ignored. It's a mess. 

Can we file a bug on this?